### PR TITLE
Add new xml implementation

### PR DESCRIPTION
**Pull request by SupunS**
This PR adds the new XML implementation.

Below is a sample with the new XML initialization and the new native functions:
```
import ballerina.lang.system;
import ballerina.lang.xmls;

function main (string[] args) {
    xml x1 = xmls:parse("<ns0:name xmlns:ns0=\"http://sample.com/test\"><ns0:fname>supun</ns0:fname><!-- two tags with 'fname' --><ns0:fname>thilina</ns0:fname><ns0:lname>setunga</ns0:lname></ns0:name>");

    xml x2 = xmls:parse("<ns1:address xmlns:ns1=\"http://sample.com/test\"><country>SL</country><city>Colombo</city></ns1:address>");

    xml x3 = xmls:parse("<root/>");
    system:print("Is Singleton: ");
    system:println(xmls:isSingleton(x1));
    system:print("Is Empty: ");
    system:println(xmls:isSingleton(x1));
    system:print("Element Name: ");
    system:println(xmls:getElementName(x1));
    system:print("Element Type: ");
    system:println(xmls:getItemType(x1));

    xml concatSeq = x1 + x2;
    xmls:setChildren(x3, x1+x2);

    xml elements = xmls:elements(concatSeq);
    system:println(elements);

    xml selectedElements = xmls:select(concatSeq, "{http://sample.com/test}name");
    system:println(selectedElements);

    xml children = xmls:children(x1);
    system:println(children);

    xml selectedChildren = xmls:selectChildren(x1, "{http://sample.com/test}fname");
    system:println(selectedChildren);

    system:println(concatSeq);

    xml copiedSeq = xmls:copy(concatSeq);
    system:println(copiedSeq);
}
```

Accessing attributes:
```
    xml book = xmls:parse("<book/>");
    xmls:setAttribute(book, "{http://sample.com/a}id", "ns1", "1234");
    xmls:setAttribute(book, "id", "", "789");

    system:println(xmls:getAttribute(book, "{http://sample.com/a}id"));
    system:println(xmls:getAttribute(book, "id"));
```

### CODE REVIEW

File: modules/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BLangVM.java [hunk 1]
```diff
@@ -644,6 +646,13 @@ private void exec() {
                     k = operands[2];
                     sf.stringRegs[k] = sf.stringRegs[i] + sf.stringRegs[j];
                     break;
+                case InstructionCodes.RADD:
```
Comment (sameera-jayasoma) [inline modules/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BLangVM.java]: Shall we make this XMLADD?

Comment (sameera-jayasoma) [inline modules/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BLangVM.java]: RADD too generic

File: modules/ballerina-native/src/main/ballerina/ballerina/lang/xmls/natives.bal [hunk 2]
```diff
@@ -98,3 +98,35 @@ native function toString (xml x) (string);
 @doc:Param { value:"value: Attribute value" }
 native function addAttribute (xml x, string xPath, string name, string value);
 
+native function isSingleton(xml x) (boolean);
```
Comment (sameera-jayasoma) [inline modules/ballerina-native/src/main/ballerina/ballerina/lang/xmls/natives.bal]: Require adequate docs :)

Comment (SupunS) [inline modules/ballerina-native/src/main/ballerina/ballerina/lang/xmls/natives.bal]: Added

File: modules/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BLangVM.java (hunk 1 above)
Comment (SupunS) [inline modules/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BLangVM.java]: Changed to XMLADD

**Pull request merged**